device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/auth/IotHubSasTokenAuthenticationProvider.java

```diff
@@ -120,7 +120,7 @@ public boolean shouldRefreshToken(boolean proactivelyRenew)
 
     public int getMillisecondsBeforeProactiveRenewal()
     {
-        return (int) this.tokenValidSecs * this.timeBufferPercentage;
```

Comment:
This was a bug where the token renewal time was much more proactive than it needed to be. Note that it was doing (time in seconds) X (85) which yielded a unit of time between seconds and milliseconds. But the return value was supposed to be in milliseconds

Reply:
Won't the value be 10x larger now?

Is it a count down or count up?

Reply:
@timtay-microsoft ?

Reply:
The value will be 10 times larger, yes. The bug here was that the sas token was previously being renewed at 8.5% of its life spent, instead of the intended 85%. This was due to an incorrect mixing of seconds and milliseconds, and the timeBufferPercentage being treated as 85 instead of .85

Reply:
Please add code comments.